python/tskit/trees.py

```diff
@@ -1600,6 +1600,36 @@ def newick(self, precision=14, root=None, node_labels=None):
             return self.__build_newick(root, precision, node_labels) + ";"
         return s
 
+    def as_dict_of_dicts(self):
+        """
+        Convert tree to dict of dicts for conversion to a
+        `networkx graph <https://networkx.github.io/documentation/stable/
+        reference/classes/digraph.html>`_.
+
+        For example::
+
+            >>> import networkx as nx
+            >>> nx.DiGraph(tree.as_dict_of_dicts())
+            >>> # undirected graphs work as well
+            >>> nx.Graph(tree.as_dict_of_dicts())
+
+        :return: Dictionary of dictionaries of dictionaries where the first key
+            is the source, the second key is the target of an edge, and the
+            third key is an edge annotation. At this point the only annotation
+            is the difference in "time" of source and target (in generations).
+        """
+        tables = self.tree_sequence.dump_tables()
```

Comment:
I don't think we need to dump the tables here. How about
```python
dod = {}
for u in tree.nodes():
    dod[u] = something(tree.children(u))
    # Store the node, including the branch length?
```

Reply:
Sure!  That sounds a lot better.